checkov/terraform/graph_builder/local_graph.py

```diff
@@ -139,6 +145,85 @@ def _add_block_data_to_graph(self, idx: int, block: TerraformBlock) -> None:
+                                provider_name = self._get_the_default_provider(self.module.temp_tf_definition.get(TFDefinitionKey(path_for_tf_definition.tf_source_modules.path), {}).get(BlockType.PROVIDER, []), path_for_tf_definition, module=module)
+                                self._assign_provider_fields(vertex, provider_name)
+                            else:
+                                module = [m for m in self.module.temp_tf_definition.get(TFDefinitionKey(path_for_tf_definition.tf_source_modules.path)).get(BlockType.MODULE) if list(m.keys())[0] == vertex.source_module_object.name]  # type:ignore
+                                provider_name = self._get_the_default_provider(self.module.temp_tf_definition.get(TFDefinitionKey(path_for_tf_definition.tf_source_modules.path), {}).get(BlockType.PROVIDER, []), path_for_tf_definition, module=module)
+                                self._assign_provider_fields(vertex, provider_name)
+                            break
+                        else:
+                            path_for_tf_definition = TFDefinitionKey(tf_source_modules=path_for_tf_definition.tf_source_modules.nested_tf_module, file_path=path_for_tf_definition.tf_source_modules.path)
+
+    @staticmethod
+    def _assign_provider_fields(vertex: TerraformBlock, provider_name: str) -> None:
+        vertex.attributes[CustomAttributes.PROVIDER_ADDRESS] = provider_name
+        v_name = vertex.name.split('.')
+        vertex.config[v_name[0]][v_name[1]][CustomAttributes.PROVIDER_ADDRESS] = provider_name
+
+    def _get_provider_address(self, path_for_tf_definition: TFDefinitionKeyType) -> list[int] | None:
+        if isinstance(path_for_tf_definition, TFDefinitionKey):
+            return self.vertices_by_module_dependency[path_for_tf_definition.tf_source_modules].get(BlockType.PROVIDER)
+        return None
+
+    def _get_the_default_provider(
+            self,
+            providers: list[dict[str, dict[str, Any]] | str],
+            path_for_tf_definition: TFDefinitionKeyType,
+            module: Optional[list[dict[str, Any]]] = None,
+            is_same_file: bool = False
+    ) -> str:
+        provider_address = self._get_provider_address(path_for_tf_definition)
+        if module and len(module) > 0:
+            module_providers = module[0][list(module[0].keys())[0]].get('providers', [''])[0]
+            if module_providers:
+                for _, m_alias in module_providers.items():
+                    if not provider_address:
+                        return cast(str, module_providers[list(module_providers.keys())[0]].replace("$", "").replace("{", "").replace("}", ""))
+                    else:
+                        for p_address in provider_address:
+                            if m_alias.replace("$", "").replace("{", "").replace("}", "") == self.vertices[p_address].name:
```

Comment:
can you extract those strings to consts? you use them multiple times

Reply:
Done